### PR TITLE
create pinot-integration-test-base module test-jar 

### PR DESCRIPTION
**Pull request by walterddr**
## Description

Split the pinot-integration-tests into a test jar util that can be imported from other 3rd party projects.
This would make it super easy to spin up pinot testing environment for ITCase.

## Upgrade Notes
N/A

Does this PR fix a zero-downtime upgrade introduced earlier?
N/A

Does this PR otherwise need attention when creating release notes? Things to consider:
N/A

## Documentation
Maybe?

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Might be cleaner if we name the new package `pinot-integration-test-base`?

**Comment by walterddr** [comment]
> Might be cleaner if we name the new package `pinot-integration-test-base`?

yes. that sounds like a better name, making the change when resolving the merge conflicts

File: pinot-integration-test-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+      </plugin>
+    </plugins>
+  </build>
+
+  <!-- cloud.localstack:localstack-utils brings in slightly older versions of all of
+       these libraries, so we have to manage their versions -->
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>software.amazon.awssdk</groupId>
```
Comment (xiangfu0) [inline pinot-integration-test-base/pom.xml:69]: shall we only keep pinot and common jars here and remove awssdk ?

**Comment by kishoreg** [comment]
@kkrugler was looking for something like this. Let's document how one can use this to test pinot.

**Comment by walterddr** [comment]
hmm couldn't reproduce this issue locally by running the compatibility check script in .github folder. any thoughts: @mcvsubbu ?
I was wondering if we can provide downloading of the artifact of the working directory so that we can debug easier. it should be pretty easy to add.

**Comment by walterddr** [comment]
This PR should be good for review. took me a while to fix the compatibility verifier. 
Any thoughts on the structure of the test-jar? @kkrugler @xiangfu0

File: pinot-integration-test-base/pom.xml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      <groupId>cloud.localstack</groupId>
+      <artifactId>localstack-utils</artifactId>
+      <version>${localstack-utils.version}</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>software.amazon.awssdk</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (xiangfu0) [inline pinot-integration-test-base/pom.xml:188]: do we need this dependency ?

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**